stack/stack_verify.cpp

```diff
@@ -0,0 +1,54 @@
+#include "stack_verify.h"
+
+void stack_dump(my_stack *stk) {
+    printf("adress of 1st elem of stk: %p\n", stk->data);
+    printf("size of stk: %d\n", stk->size - 1);
+    printf("capasity of stk: %d\n", stk->capacity);
+    printf("stk wo canary: ");
+    for (int i = 1; i < stk->size; i++) {
+        if (stk->data[i] != CANARY_VALUE) {
+            printf("%f ", stk->data[i]);
+        }
+    }
+    printf("\n");
+    printf("poisoned value (index in stk = %d) %f\n", stk->size - 1, stk->data[stk->size]);
+    printf("another poisoned value (index in stk = %d) %f\n", stk->capacity - 2 - 1, stk->data[stk->capacity - 2]);
+    if (!isnan(stk->popped)){
```

Comment:
add general check of poisoned value of type `stackelem_t`

Reply:
resolved